test/string.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <unity.h>
+
+static void
+test_strcmp(void)
+{
+	/* Test with long string. */
+	TEST_ASSERT_EQUAL(0, strcmp("Here is a string", "Here is a string"));
+
+	/* Test with medium string. */
+	TEST_ASSERT_EQUAL(0, strcmp("Hello World", "Hello World"));
+
+	/* Test with short string. */
+	TEST_ASSERT_EQUAL(0, strcmp("A", "A"));
+
+	/* Test with numbers in string. */
+	TEST_ASSERT_EQUAL(0, strcmp("He11o W0rld", "He11o W0rld"));
+
+	/* Test with special characters. */
+	TEST_ASSERT_EQUAL(0, strcmp("Hello_World!", "Hello_World!"));
+
+	/* Test string with only a space. */
+	TEST_ASSERT_EQUAL(0, strcmp(" ", " "));
+
+	/* Test an empty string. */
+	TEST_ASSERT_EQUAL(0, strcmp("", ""));
+
+	/* Test two different strings. */
+	int result = strcmp("Here is a string", "Here is a different string");
+	TEST_ASSERT(result == 15);
```

Comment:
These are standard functions. If you want to write tests for them, you need to look at their documentation. From the POSIX manual page:

> Upon  completion,  strcmp()  shall  return  an integer greater than, equal to, or less than 0, if the string pointed to by s1 is greater than, equal to, or less than the string pointed to by s2,  respectively.

So you can't rely on the magnitude of the result, only its sign.